pkg/detectors/fastlypersonaltoken/fastlypersonaltoken_test.go

```diff
@@ -1,120 +1,92 @@
-//go:build detectors
-// +build detectors
-
 package fastlypersonaltoken
 
 import (
 	"context"
-	"fmt"
 	"testing"
-	"time"
 
-	"github.com/kylelemons/godebug/pretty"
+	"github.com/google/go-cmp/cmp"
+
 	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/engine/ahocorasick"
+)
 
-	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
-	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+var (
+	// example picked from: https://github.com/ryan-miller/learn-go-with-tests/blob/181467c9e512f7e68d3f3cbcea89f0050982416c/fastly/users.go#L22
+	validPattern = `
+	// headers and header values
+	const fastlyKeyToken string = "Fastly-Key"
+	const fastlyKey string = "TVAWji0p7uDI6OP9DyWvmV-vgoUoXIuf"
+	const contentTypeToken string = "Content-Type"
+	const appJsonContentType = "application/json"`
+
+	invalidPattern = `
+	// headers and header values
+	const fastlyKeyToken string = "Fastly-Key"
+	const fastlyKey string = "$FASTLY_KEY"
+	const contentTypeToken string = "Content-Type"
+	const appJsonContentType = "application/json"`
 )
 
-func TestFastlyPersonalToken_FromChunk(t *testing.T) {
-	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
-	defer cancel()
-	testSecrets, err := common.GetSecret(ctx, "trufflehog-testing", "detectors3")
-	if err != nil {
-		t.Fatalf("could not get test secrets from GCP: %s", err)
-	}
-	secret := testSecrets.MustGetField("FASTLYPERSONALTOKEN_TOKEN")
-	inactiveSecret := testSecrets.MustGetField("FASTLYPERSONALTOKEN_INACTIVE")
+func TestFastlyPersonalToken_Pattern(t *testing.T) {
+	d := Scanner{}
+	ahoCorasickCore := ahocorasick.NewAhoCorasickCore([]detectors.Detector{d})
 
-	type args struct {
-		ctx    context.Context
-		data   []byte
-		verify bool
-	}
 	tests := []struct {
-		name    string
-		s       Scanner
-		args    args
-		want    []detectors.Result
-		wantErr bool
+		name  string
+		input string
+		want  []string
 	}{
 		{
-			name: "found, verified",
-			s:    Scanner{},
-			args: args{
-				ctx:    context.Background(),
-				data:   []byte(fmt.Sprintf("You can find a fastlypersonaltoken secret %s within", secret)),
-				verify: true,
-			},
-			want: []detectors.Result{
-				{
-					DetectorType: detectorspb.DetectorType_FastlyPersonalToken,
-					Verified:     true,
-				},
-			},
-			wantErr: false,
+			name:  "valid pattern",
+			input: validPattern,
+			want:  []string{"TVAWji0p7uDI6OP9DyWvmV-vgoUoXIuf"},
```

Comment:
Can we pass on validPattern variable in `want`? It will consolidate valid token in one variable, update will become easy.

Reply:
ValidPattern is actually a long testing string, I can create a separate variable with output.